### PR TITLE
Networking Phase 2: Code commenting and optimization and a fix

### PR DESCRIPTION
**Pull request by BaronGreenback**
No new functionality.

Fixed: DLNA Debug output is now unescaped and html decoded, making it actually useful!

Added comments to dlna didl class.
Some static methods as recommended.
Added Utf8StringWriter to replace StringWriterWithEncoding.cs.
Got rid of a TODO.

### CODE REVIEW

File: Emby.Dlna/Didl/DidlBuilder.cs [hunk 1]
```diff
@@ -94,36 +125,49 @@ public string GetItemDidl(BaseItem item, User user, BaseItem context, string dev
...
+
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
             foreach (var att in profile.XmlRootAttributes)
             {
-                var parts = att.Name.Split(':', StringSplitOptions.RemoveEmptyEntries);
+                var parts = att.Name.Split(new[] { ':' }, StringSplitOptions.RemoveEmptyEntries);
```
Comment (Bond-009) [inline Emby.Dlna/Didl/DidlBuilder.cs:170]: Why?

Comment (BaronGreenback) [reply Emby.Dlna/Didl/DidlBuilder.cs]: Bad merge. have fixed.

File: Emby.Dlna/Didl/DidlBuilder.cs [hunk 2]
```diff
@@ -952,6 +1124,14 @@ private void AddAlbumArtist(XmlWriter writer, string name)
             }
         }
 
+        /// <summary>
+        /// Adds a Value.
+        /// </summary>
+        /// <param name="writer">The <see cref="XmlWriter"/>.</param>
+        /// <param name="prefix">The <see cref="string"/>.</param>
+        /// <param name="name">The <see cref="string"/>.</param>
+        /// <param name="value">The <see cref="string"/>.</param>
```
Comment (crobibero) [inline Emby.Dlna/Didl/DidlBuilder.cs:1133]: Auto comment tools are sometimes bad

Comment (BaronGreenback) [reply Emby.Dlna/Didl/DidlBuilder.cs]: Manually gone through and corrected.

**Review by crobibero — APPROVED** [review]

**Comment by BaronGreenback** [comment]
Migrated across to plugins

**Pull request closed without merging**